util/topsql/reporter/datamodel.go

```diff
@@ -497,10 +497,46 @@ func (c *collecting) appendOthersStmtStatsItem(timestamp uint64, item stmtstats.
 	others.appendStmtStatsItem(timestamp, item)
 }
 
+// removeInValidPlanRecord remove "" plan if there are only 1 valid plan in the record.
+func (c *collecting) removeInValidPlanRecord() {
+	sql2PlansMap := make(map[string][][]byte, len(c.records)) // sql_digest => []plan_digest
+	for _, v := range c.records {
+		k := string(v.sqlDigest)
+		sql2PlansMap[k] = append(sql2PlansMap[k], v.planDigest)
+	}
+	for k, plans := range sql2PlansMap {
+		if len(plans) != 2 {
```

Comment:
Is this reasonable? Do we have only two `Plans` for the same `SQL`?

Reply:
No, but We can only remove the `""` plan when there are only 1 valid plan, so the `len(plans) == 2` is a filter to filter the SQL which `len(plans) == 2` and next filter use to filter at least 1 plan is `""` plan.